test/gurka/test_closure_annotations.cc

```diff
@@ -0,0 +1,327 @@
+#include "gurka.h"
+#include "test.h"
+
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+using LiveTrafficCustomize = test::LiveTrafficCustomize;
+
+namespace {
+
+inline void SetLiveSpeedFrom(baldr::TrafficSpeed* live_speed, uint64_t speed, uint8_t subsegment) {
+  live_speed->breakpoint1 = subsegment;
+  live_speed->breakpoint2 = 255;
+  live_speed->speed2 = speed >> 1;
+  live_speed->speed3 = UNKNOWN_TRAFFIC_SPEED_RAW;
+}
+
+inline void SetLiveSpeedUpto(baldr::TrafficSpeed* live_speed, uint64_t speed, uint8_t subsegment) {
```

Comment:
same note about name here

Reply:
The speed datatype should probably be uint8_t too

Reply:
How about `overall_speed`?

Reply:
`SetLiveSpeed` below can probably be collapsed into this function if you set `subsegment/breakpoint1` argument to default to 255?

Reply:
I'm not sure what `overall_speed` should be set to if its already populated (I do that when constructing the gurka map in the setup function) and we want to close only part of the edge. Thats why I left it as is

Reply:
The most important is probably to set `overall_speed` to 0 if any of the subsegments have zero speed. That marks the edge as closed if a subsegment is closed. Apart from that, it's not used if you have more than one subsegment.